oneadapter/src/main/java/com/idanatz/oneadapter/internal/holders_creators/ViewHolderCreatorsStore.kt

```diff
@@ -11,7 +11,7 @@ internal class ViewHolderCreatorsStore {
         holderCreators[clazz] = creator
     }
 
-    fun getCreatorUniqueIndex(clazz: Class<Diffable>): Int {
```

Comment:
What is the reason for this change?
Each item type the adapter holds must implement Diffable, it doesn't matter if its an internal holder type or external class from the user.

Reply:
The reason I changed this is to allow calling the function like this:

```kotlin
getCreatorUniqueIndex(LoadingIndicator::class.java)
```

When the type is < Diffable > rather then <*> the compiler complains that LoadingIndicator is not a Diffable even though it is. This does work with instances but not when using ``` LoadingIndicator::class.java ```. You could do a runtime check maybe and throw if it doesn't inherit from Diffable.

Reply:
ViewHolderCreatorsStore is an internal class of the library, it should not be used with classes other than DIffable.

The way i see it, the function on OneAdapter.kt can be called with type < * >, but inside InternalAdapter, the function getItemViewTypeFromClass will validate the class if it is of type DIffable (using the Validator class) and throw a new exception (that will be written in the Exceptions.kt)

class UnsupportedClassException(msg: String = "Class must implement Diffable interface") : Throwable(msg)

Reply:
I pushed the changes, let me know what u think :)